### PR TITLE
nixos/tomcat: configure default group and fix broken default package …

### PR DESCRIPTION
**Pull request by svanderburg**
Without this fix, evaluating a NixOS configuration with Tomcat enabled and the
default settings results in the following evaluation error:

Failed assertions:
- users.users.tomcat.group is unset. This used to default to
nogroup, but this is unsafe. For example you can create a group
for this user with:
users.users.tomcat.group = "tomcat";
users.groups.tomcat = {};

###### Description of changes

Added default group property and changed the broken default package reference

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-servers/tomcat.nix [hunk 1]
```diff
@@ -23,7 +23,7 @@ in
 
       package = mkOption {
         type = types.package;
-        default = pkgs.tomcat85;
+        default = pkgs.tomcat9;
         defaultText = literalExpression "pkgs.tomcat85";
```
Comment (pennae) [inline nixos/modules/services/web-servers/tomcat.nix:27]: this line also has to be changed to match

Comment (svanderburg) [reply nixos/modules/services/web-servers/tomcat.nix]: Ah yes, good catch. I just did a grep on anything referring to `tomcat8` and found another bad example. I've just corrected it and amended the commit.

File: nixos/modules/services/web-servers/tomcat.nix [hunk 2]
```diff
@@ -127,7 +127,7 @@ in
       webapps = mkOption {
         type = types.listOf types.path;
         default = [ tomcat.webapps ];
-        defaultText = literalExpression "[ pkgs.tomcat85.webapps ]";
+        defaultText = literalExpression "[ pkgs.tomcat9.webapps ]";
```
Comment (pennae) [inline nixos/modules/services/web-servers/tomcat.nix:130]:
```suggestion
        defaultText = literalExpression "[ config.services.tomcat.package.webapps ]";
```
is probably better here than the text that was there originally. the old text gave the impression that changing `package` did not also change where this option pulls its value.

Comment (svanderburg) [reply nixos/modules/services/web-servers/tomcat.nix]: That's indeed not a bad suggestion. I've just amended the commit again :)

**Comment by svanderburg** [comment]
I've noticed that there's no test coverage for tomcat, so I've added a basic test suite that checks if the example app works

**Comment by svanderburg** [comment]
If there are no further comments, I'll merge this tonight

**Pull request merged**